cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/resource/AzureAttachmentResourceBuilder.java

```diff
@@ -47,23 +50,30 @@
                 .findFirst()
                 .orElseThrow(() -> new AzureResourceException("Instance resource not found"));
 
+        CloudContext cloudContext = auth.getCloudContext();
+        String resourceGroupName = azureResourceGroupMetadataProvider.getResourceGroupName(cloudContext, cloudStack);
+        AzureClient client = getAzureClient(auth);
+        VirtualMachine vm = client.getVirtualMachine(resourceGroupName, instance.getName());
+        Set<String> diskIds = vm.dataDisks().values().stream().map(VirtualMachineDataDisk::id).collect(Collectors.toSet());
+
         CloudResource volumeSet = buildableResource.stream()
                 .filter(cloudResource -> cloudResource.getType().equals(ResourceType.AZURE_VOLUMESET))
                 .filter(cloudResource -> !instance.getInstanceId().equals(cloudResource.getInstanceId()))
                 .findFirst()
                 .orElseThrow(() -> new AzureResourceException("Volume set resource not found"));
 
-        CloudContext cloudContext = auth.getCloudContext();
-        String resourceGroupName = azureResourceGroupMetadataProvider.getResourceGroupName(cloudContext, cloudStack);
-        AzureClient client = getAzureClient(auth);
 
         VolumeSetAttributes volumeSetAttributes = getVolumeSetAttributes(volumeSet);
         volumeSetAttributes.getVolumes()
                 .forEach(volume -> {
-                            Disk disk = client.getDiskById(volume.getId());
-                            VirtualMachine vm = client.getVirtualMachine(resourceGroupName, instance.getName());
-                            client.attachDiskToVm(disk, vm);
-                        });
+                    Disk disk = client.getDiskById(volume.getId());
+                    if (!diskIds.contains(disk.id())) {
```

Comment:
Can you add some log lines please that we're detaching the disk or fi we're not doing anything as it's already attached.

Reply:
Detaching has the log in detachDiskFromVm() method, attaching in attachDiskToVm()